runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MonitoringInfoMetricName.java

```diff
@@ -112,12 +135,27 @@ public boolean equals(Object o) {
     return this.urn.equals(other.urn) && this.labels.equals(other.labels);
   }
 
+  /**
+   * String-representation of this {@link MonitoringInfoMetricName} with a custom delimiter.
+   *
+   * <p>See discussion on {@link MetricKey#toString}; metric "label" info is currently split between
+   * {@link MetricKey#stepName} and {@link MonitoringInfoMetricName#labels} (the former holds {@link
+   * SimpleMonitoringInfoBuilder#PTRANSFORM_LABEL ptransform} info, and the latter {@link
+   * SimpleMonitoringInfoBuilder#PCOLLECTION_LABEL info}.
+   *
+   * <p>For now, we bake in the assumption here that {@link MonitoringInfoMetricName#toString} is
+   * only responsible for inlining the pcollection label to the string representation.
+   *
+   * <p>Later, {@link MonitoringInfoMetricName} and {@link MetricKey} will be merged, and this
+   * handling unified.
+   */
   @Override
-  public String toString() {
-    StringBuilder builder = new StringBuilder();
-    builder.append(this.urn.toString());
-    builder.append(" ");
-    builder.append(this.labels.toString());
-    return builder.toString();
+  public String toString(String delimiter) {
+    StringBuilder sb = new StringBuilder();
+    if (labels.containsKey(PCOLLECTION_LABEL)) {
```

Comment:
I really don't understand this decision, just make it print all the labels. Why special case PCOLLECTION_LABEL